commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -0,0 +1,357 @@
+     *
+     * @param minCapacity – Desired minimum capacity.
+     * @return the backing double array.
+     * @throws OutOfMemoryError - if the {@code minCapacity} is greater than {@code MAX_ARRAY_SIZE}.
+     */
+    private double[] ensureCapacityInternal(int minCapacity) {
+        modCount++;
+        final long minArrayCapacity = minCapacity * 2L;
+        if (minArrayCapacity > MAX_ARRAY_SIZE) {
+            throw new OutOfMemoryError(String.format(OOM_ERROR_STRING, minArrayCapacity));
+        }
+        final long oldArrayCapacity = realAndImagParts.length;
+        if (minArrayCapacity > oldArrayCapacity) {
+            long newArrayCapacity = oldArrayCapacity + (oldArrayCapacity >> 1);
+            if (newArrayCapacity % 2 != 0) {
+                ++newArrayCapacity;
+            }
+            if (newArrayCapacity > MAX_ARRAY_SIZE) {
+                newArrayCapacity = MAX_ARRAY_SIZE;
+            } else if (newArrayCapacity < minArrayCapacity) {
+                newArrayCapacity = minArrayCapacity;
+            }
+            realAndImagParts = Arrays.copyOf(realAndImagParts, (int)newArrayCapacity);
+        }
+        return realAndImagParts;
+    }
+
+    /**
+     * Increases the capacity of this ComplexList instance, if necessary, to ensure that it can hold at
+     * least an additional number of elements specified by the capacity argument.
+     *
+     * @param capacity - Desired capacity.
+     */
+    private void expand(int capacity) {
+        ensureCapacityInternal(size + capacity);
+    }
+
+    /**
+     * Appends the specified complex element to the end of this list.
+     *
+     * @param element - Complex element to be appended to this list.
+     * @return true after element has been added and size has been updated.
+     */
+    @Override
+    public boolean add(Complex element) {
+        double[] e = realAndImagParts;
+        if (size == (e.length >>> 1)) {
+            e = ensureCapacityInternal(size + 1);
+        }
+        final int i = size << 1;
+        e[i] = element.real();
+        e[i + 1] = element.imag();
+        size++;
+        return true;
+    }
+
+    /**
+     * Inserts the specified element at the specified position in this list. Shifts the element currently at that
+     * position (if any) and any subsequent elements to the right (adds one to their indices).
+     *
+     * @param index – Index at which the specified element is to be inserted.
+     * @param element – Complex element to be inserted.
+     * @throws IndexOutOfBoundsException – if index isn't within the range of list.
+     */
+    @Override
+    public void add(int index, Complex element) {
+        rangeCheckForInsert(index);
+        double[] e = realAndImagParts;
+        if (size == e.length >>> 1) {
+            e = ensureCapacityInternal(size + 1);
+        }
+        final int i = index << 1;
+        System.arraycopy(e, 2 * index, e, i + 2, (size * 2) - i);
```

Comment:
You already have `2 * index` so you can reuse `i` here. For consistency perhaps use `size << 1` as well.